Assets/Scripts/PlayerController.cs

```diff
@@ -21,14 +21,14 @@ private void SetRandomPlayerColor()
 
     private void OnCollisionEnter(Collision collision) {
         var coliderTag = collision.gameObject.tag;
-        if(coliderTag== currentPlayerColor.enemyColorTag){
+        if(coliderTag== currentPlayerColor.enemyData.enemyTag){
             var id = collision.gameObject.GetComponent<EnemyController>().GetId();
             GameEvents.current.EnemeyDestroyRequred(id);
             SetRandomPlayerColor();
         }else{
             for (int i = 0; i < allEnemysAndColors.Length; i++)
             {
-                if(coliderTag == allEnemysAndColors[i].enemyColorTag){
+                if(coliderTag == allEnemysAndColors[i].enemyData.enemyTag){
                     GameEvents.current.PlayerDestroyRequred();
                 }
             }

```

Comment:
And here collision magic, don't need to write Tag's and stuff anymore, now I think I also don't need determinate even color :DD will update that soon